### PR TITLE
OpenStack: update gophercloud package and add new auth options

### PR DESCRIPTION
**Pull request by kayrus**
## Description

This PR adds new auth options into the openstack client and removes deprecated keystone V2 support.

Additionally it adds an ability to remove the VM, which was not created successfully. If it is not removed, new VMs cannot be created, because the resources quota can be reached.

## Related issue(s)

Resolves https://github.com/rancher/rancher/issues/15796

/cc @daxmc99 @loganhz @dbagg

### CODE REVIEW

**Comment by kayrus** [comment]
@daxmc99 I split the PR into 4 commits no by an accident, but by a purpose. Is it an option to review each commit independently?

**Comment by daxmc99** [comment]
Can you explain why you need go modules for this fix? We are currently working on moving to go modules but we have a few dependencies that will need to ensure will not change. You should be fine using dep for what you are changing.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	google.golang.org/cloud v0.0.0-20151119220103-975617b05ea8 // indirect
+	gopkg.in/airbrake/gobrake.v2 v2.0.9 // indirect
+	gopkg.in/gemnasium/logrus-airbrake-hook.v2 v2.1.2 // indirect
+	gopkg.in/yaml.v2 v2.2.4
+	launchpad.net/gocheck v0.0.0-20140225173054-000000000087 // indirect
+)
+
+replace (
+	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.0.4
+	github.com/sirupsen/logrus => github.com/sirupsen/logrus v1.4.2
```
Comment (daxmc99) [inline go.mod]: I haven't seen this before? Why do these need to point towards different versions?

**Comment by daxmc99** [comment]
We moved to go modules, could you rebase your PR?

**Comment by kayrus** [comment]
@daxmc99 sorry for not responding, I just noticed your comments. I rebased the PR.

File: go.mod (hunk 1 above)
Comment (kayrus) [reply go.mod]: I brought this from the upstream docker machine patch. It had a dependency conflict between an old repo name and version, which didn't support go modules and a new version, which supports go modules.

**Comment by daxmc99** [comment]
Can you point out where you remove keystone v2 support? I would prefer to not break backwards compatibility.

**Comment by kayrus** [comment]
@daxmc99 identity v2 is not removed. An `openstack.Authenticate` method automatically detects whether it should auth against v2 or v3, see https://github.com/gophercloud/gophercloud/blob/7aa2e52d21f9c621caf111307a1e3590884abc47/openstack/client.go#L89

**Comment by kayrus** [comment]
@daxmc99 Does it need 3 approvals to get merged?

**Pull request merged**